src/routes/ChatRoom/ChatRoom.js

```diff
@@ -0,0 +1,514 @@
+import cx from 'classnames'
+import { debounce, trim } from 'lodash/fp'
+import moment from 'moment-timezone'
+import { EditorView } from 'prosemirror-view'
+import React, { useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState } from 'react'
+import { IoSend } from 'react-icons/io5'
+import ReactResizeDetector from 'react-resize-detector'
+import { Virtuoso } from 'react-virtuoso'
+import AttachmentManager from 'components/AttachmentManager'
+import Button from 'components/Button'
+import HyloEditor from 'components/HyloEditor'
+import Icon from 'components/Icon'
+import LinkPreview from 'components/PostEditor/LinkPreview'
+import Loading from 'components/Loading'
+import NoPosts from 'components/NoPosts'
+import PostCard from 'components/PostCard'
+import TopicFeedHeader from 'components/TopicFeedHeader'
+import UploadAttachmentButton from 'components/UploadAttachmentButton'
+import { MAX_POST_TOPICS } from 'util/constants'
+import isWebView from 'util/webView'
+import ChatPost from './ChatPost'
+
+import styles from './ChatRoom.scss'
+
+// Hack to fix focusing on editor after it unmounts/remounts
+EditorView.prototype.updateState = function updateState (state) {
+  if (!this.docView) return // This prevents the matchesNode error on hot reloads
+  this.updateStateInner(state, this.state.plugins !== state.plugins)
+}
+
+// the maximum amount of time in minutes that can pass between messages to still
+// include them under the same avatar and timestamp
+const MAX_MINS_TO_BATCH = 5
+
+// Recommended here:
+// https://github.com/ueberdosis/tiptap/issues/2403#issuecomment-1062712162
+function useEventCallback (fn) {
+  const ref = useRef()
+  useLayoutEffect(() => {
+    ref.current = fn
+  })
+  return useCallback(() => (0, ref.current)(), [])
+}
+
+export default function ChatRoom (props) {
+  const {
+    // changeSearch,
```

Comment:
commented out for now? I don't know if search will be a feature for chat

Reply:
yeah I could delete, not sure if we will add search to chat rooms either